internal/appsec/waf.go

```diff
@@ -130,25 +129,23 @@ func newHTTPWAFEventListener(handle *waf.Handle, addresses []string, timeout tim
 			return
 		}
 
-		// OnUserIDOperationStart happens when appsec.SetUser() is called. We run the WAF and apply actions to
-		// see if the associated user should be blocked. Since we don't control the execution flow in this case
-		// (SetUser is SDK), we delegate the responsibility of interrupting the handler to the user.
-		op.On(sharedsec.OnUserIDOperationStart(func(operation *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
-			if !addressesContains(addresses, userIDAddr) {
-				return
-			}
-
-			matches, actionIds := runWAF(wafCtx, map[string]interface{}{userIDAddr: args.UserID}, timeout)
-			if len(matches) > 0 {
-				for _, id := range actionIds {
-					if actionHandler.Apply(id, op) {
-						operation.Error = sharedsec.NewUserMonitoringError("Request blocked")
+		if _, ok := httpAddresses[userIDAddr]; ok {
```

Comment:
The test is done on the global list of HTTP addresses rather than the local list of addresses present in the rules `addresses`.
Same comment on the other conditions you added.

Feel free to revive `addressesContains` to simplify the `if` a bit.

Reply:
But this condition is in `newHTTPWAFEventListener` so it's okay maybe ?